ring/ring.go

```diff
@@ -191,6 +191,13 @@ type Ring struct {
 	// then this value will be 0.
 	oldestRegisteredTimestamp int64
 
+	readOnlyInstancesUpdated bool // True, if readOnlyInstances was actually updated. False if 0 is because of default value.
```

Comment:
Could we get rid of this field and just initialize `readOnlyInstances` to something like `-1` to indicate that it hasn't been properly set yet?

Reply:
Problem is that many tests don't initialize all the fields today, which could leave `readOnlyInstances` default value 0 being used as "no read only instances", which may not be true. I think we should fix how ring is created (with all fields initialized) and update ring fields from ringDesc in a common way, then we can remove this field too. WDYT?

Reply:
That sounds fine. You're thinking a followup PR?

Reply:
Good idea... using pointers will make compiler to check for correct usage (pointer check first).

Reply:
Implemented in https://github.com/grafana/dskit/pull/565/commits/b99a6cd9e1202b9e5cd5cadeea2b6c0c48c7ddf6.